src/main/java/org/jitsi/jibri/service/impl/FileRecordingJibriService.kt

```diff
@@ -0,0 +1,158 @@
+     */
+    private var processMonitorTask: ScheduledFuture<*>? = null
+
+    init {
+        sink = FileSink(recordingOptions.recordingDirectory, recordingOptions.callParams.callUrlInfo.callName)
+        jibriSelenium.addStatusHandler {
+            publishStatus(it)
+        }
+    }
+
+    /**
+     * @see [JibriService.start]
+     */
+    override fun start(): Boolean {
+        if (!jibriSelenium.joinCall(recordingOptions.callParams.callUrlInfo.callName)) {
+            logger.error("Selenium failed to join the call")
+            stop()
+            return false
+        }
+        capturer.start(sink)
+        var numRestarts = 0
+        val processMonitor = ProcessMonitor(capturer) { exitCode ->
+            if (exitCode != null) {
+                logger.error("Capturer process is no longer healthy.  It exited with code $exitCode")
+            } else {
+                logger.error("Capturer process is no longer healthy but it is still running, stopping it now")
+                capturer.stop()
+            }
+            if (numRestarts == FFMPEG_RESTART_ATTEMPTS) {
+                logger.error("Giving up on restarting the capturer")
+                publishStatus(JibriServiceStatus.ERROR)
+                stop()
+            } else {
+                numRestarts++
+                // Re-create the sink here because we want a new filename
+                sink = FileSink(recordingOptions.recordingDirectory, recordingOptions.callParams.callUrlInfo.callName)
+                capturer.start(sink)
+            }
+        }
+        processMonitorTask = executor.scheduleAtFixedRate(processMonitor, 30, 10, TimeUnit.SECONDS)
+        return true
+    }
+
+    /**
+     * @see [JibriService.stop]
+     */
+    override fun stop() {
+        processMonitorTask?.cancel(false)
+        logger.info("Stopping capturer")
+        capturer.stop()
+        logger.info("Quitting selenium")
+        val participants = jibriSelenium.getParticipants()
+        val metadata = RecordingMetadata(participants)
+        jacksonObjectMapper()
+            .writeValue(File(recordingOptions.recordingDirectory, "metadata"), metadata)
+        jibriSelenium.leaveCallAndQuitBrowser()
+        logger.info("Finalizing the recording")
+        finalize()
+    }
+
+    /**
+     * Helper to execute the finalize script and wait for its completion.
+     * NOTE that this will block for however long the finalize script takes
+     * to complete (by design)
+     */
+    private fun finalize() {
+        try {
+            val finalizeProc = Runtime.getRuntime()
+                .exec("${recordingOptions.finalizeScriptPath} ${recordingOptions.recordingDirectory}")
+            finalizeProc.waitFor()
+            logger.info("Recording finalize script finished with exit " +
+                    "value: ${finalizeProc.exitValue()}")
+        } catch (e: IOException) {
+            logger.error("Failed to run finalize script: $e")
+        }
+    }
+}
```

Comment:
I really like the JibriService and in general how easy it is to use the two services.